sphinxext/opengraph/util.py

```diff
@@ -0,0 +1,67 @@
+from typing import Dict, Any
+from sphinx.config import Config
+from docutils import nodes
+from urllib.parse import urljoin
+from .visitor import OpenGraphVisitor
+from .html_parser import HTMLTextParser
+
+
+def make_tag(property: str, content: str) -> str:
+    return f'<meta property="{property}" content="{content}" />\n  '
+
+
+def insert_tags(
+    context: Dict[str, Any], doctree: nodes.document, config: Config
+) -> None:
+    # Set length of description
+    try:
+        desc_len = int(config["ogp_description_length"])
+    except ValueError:
+        desc_len = 300
+
+    # parse out any html from the title
+    html_parser = HTMLTextParser()
+    if context["title"] != "&lt;no title&gt;":
+        # only use title if its exists
+        html_parser.feed(context["title"])
+    html_parser.close()
```

Comment:
I know this is this original implementation. But, I think it would be better as a function.

```py
def santitize_title(title: str) -> str:
	# take in a title and return a clean version.
```

Also, `<no title>` shouldn't be an edge case. That is the page title as used by Sphinx.

Reply:
I already removed `<no title>`, haven't pushed yet.
And that function would go in `util.py` right?

Reply:
Yeah, util would be a good place for it.